compute/api/Compute.Samples/CreateInstanceAsync.cs

```diff
@@ -0,0 +1,105 @@
+        string zone = "us-central1-a",
+        string machineName = "test-machine",
+        string machineType = "n1-standard-1",
+        string diskImage = "projects/debian-cloud/global/images/family/debian-10",
+        string diskSizeGb = "10")
+    {
+        
+        Instance instance = new Instance
+        {
+            Name = machineName,
+            // See https://cloud.google.com/compute/docs/machine-types for more information on machine types.
+            MachineType = $"zones/{zone}/machineTypes/{machineType}",
+            // Instance creation requires at least one persistent disk.
+            Disks =
+            {
+                new AttachedDisk
+                {
+                    AutoDelete = true,
+                    Boot = true,
+                    Type = AttachedDisk.Types.Type.Persistent,
+                    InitializeParams = new AttachedDiskInitializeParams 
+                    {
+                        // See https://cloud.google.com/compute/docs/images for more information on available images.
+                        SourceImage = diskImage,
+                        DiskSizeGb = diskSizeGb
+                    }
+                }
+            },
+            // Instance creation requires at least one network interface.
+            // The "default" network interface is created automatically for every project.
+            NetworkInterfaces = { new NetworkInterface { Name = "default" } }
+        };
+
+        // Initialize the client that will be used to send instance related requests.
+        // You should reuse the same client for multiple requests.
+        InstancesClient client = await InstancesClient.CreateAsync();
+
+        // Make the request to create a VM instance.
+        Operation instanceCreation = await client.InsertAsync(projectId, zone, instance);
+
+        // You may poll the operation until it completes or fails, or for a given amount of time.
+        // If polling times out, the operation may still finish successfully after.
+        await PollUntilCompletedAsync(instanceCreation, projectId, zone);
+    }
+
+    private async Task PollUntilCompletedAsync(Operation operation, string projectId, string zone)
+    {
+        ZoneOperationsClient operationClient = await ZoneOperationsClient.CreateAsync();
+        GetZoneOperationRequest pollRequest = new GetZoneOperationRequest
+        {
+            Operation = operation.Name,
+            Zone = zone,
+            Project = projectId,
+        };
+
+        TimeSpan timeOut = TimeSpan.FromMinutes(3);
+        TimeSpan pollInterval = TimeSpan.FromSeconds(15);
+
+        DateTime deadline = DateTime.UtcNow + timeOut;
+        while (operation.Status != Operation.Types.Status.Done)
+        {
+            operation = await operationClient.GetAsync(pollRequest);
+
+            if (operation.Status == Operation.Types.Status.Done)
+            {
+                break;
+            }
+
+            if (DateTime.UtcNow > deadline)
```

Comment:
I note this isn't "smart" enough to realize that by the time we've finished the delay, we'll have hit the deadline... I'm hoping we'll put better retry code in the library at some point, so I'm okay with this being "simple" rather than "smart".

Reply:
The improved LRO approach for this library should be released in H2 and then we will be able to change this code to a 'smarter' one.